src/main/java/de/srendi/advancedperipherals/common/util/ChunkManager.java

```diff
@@ -2,53 +2,60 @@
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.saveddata.SavedData;
-import net.minecraftforge.common.world.ForgeChunkManager;
-import net.minecraftforge.event.TickEvent;
-import net.minecraftforge.event.server.ServerStartedEvent;
-import net.minecraftforge.event.server.ServerStoppingEvent;
-import net.minecraftforge.eventbus.api.SubscribeEvent;
-import net.minecraftforge.fml.common.Mod;
-import net.minecraftforge.server.ServerLifecycleHooks;
+import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.fml.common.Mod;
+import net.neoforged.neoforge.common.world.chunk.RegisterTicketControllersEvent;
+import net.neoforged.neoforge.common.world.chunk.TicketController;
+import net.neoforged.neoforge.event.TickEvent;
+import net.neoforged.neoforge.event.server.ServerStartedEvent;
+import net.neoforged.neoforge.server.ServerLifecycleHooks;
 import org.apache.logging.log4j.Level;
 import org.jetbrains.annotations.NotNull;
 
-import java.time.LocalDateTime;
-import java.time.ZoneOffset;
-import java.util.*;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
 
 @Mod.EventBusSubscriber(modid = AdvancedPeripherals.MOD_ID)
 public class ChunkManager extends SavedData {
 
     private static final String DATA_NAME = AdvancedPeripherals.MOD_ID + "_ForcedChunks";
     private static final String FORCED_CHUNKS_TAG = "forcedChunks";
-    private static int tickCounter = 0;
+    private static long tickCounter = 0;
     private final Map<UUID, LoadChunkRecord> forcedChunks = new HashMap<>();
     private boolean initialized = false;
 
+    private static final TicketController controller = new TicketController(AdvancedPeripherals.getRL("chunkcontroller"), null);
+
     public ChunkManager() {
         super();
     }
 
     public static @NotNull ChunkManager get(@NotNull ServerLevel level) {
-        return level.getDataStorage().computeIfAbsent(ChunkManager::load, ChunkManager::new, DATA_NAME);
+        return level.getDataStorage().computeIfAbsent(new SavedData.Factory<>(ChunkManager::new, ChunkManager::load, null), DATA_NAME);
     }
 
     public static ChunkManager load(@NotNull CompoundTag data) {
         ChunkManager manager = new ChunkManager();
         CompoundTag forcedData = data.getCompound(FORCED_CHUNKS_TAG);
+        AdvancedPeripherals.debug("Loading chunk manager from NBT " + data, Level.WARN);
         for (String key : forcedData.getAllKeys()) {
             manager.forcedChunks.put(UUID.fromString(key), LoadChunkRecord.deserialize(forcedData.getCompound(key)));
         }
         return manager;
     }
 
-    @SubscribeEvent
-    public static void beforeServerStopped(ServerStoppingEvent event) {
-        ChunkManager.get(event.getServer().overworld()).stop();
+    public static void registerTicketController(RegisterTicketControllersEvent event) {
```

Comment:
do we not need the `@SubscribeEvent` here? or I missed something

Reply:
wait I see there is `modBus.addListener(ChunkManager::registerTicketController)`

Reply:
Nope
`@SubscribeEvent` there registers the event in the forge event bus. The default bus for the annotated class `@Mod.EventBusSubscriber(modid = AdvancedPeripherals.MOD_ID)` is the forge bus, that's why it would be registered in the forge bus. But this will crash the game when neo tries to register that event with the annotation on the wrong bus.

I also can't change the bus in the annotation of the class because the other two events (`serverTick` and `afterServerStarted`) use the forge bus, not the mod bus

When you look at the specific commit, I now register the event using the modBus in the main class

Reply:
However is there any difference between this two styles? If no then can we decide to use same way to subscribe event but not mixing them together

Reply:
okay I get this, thanks for explaining.
But it will be better if you can add a comment to say it's registered in the mod event bus

Reply:
I am not sure what the best way is there, but I see a lot of mods doing it like this too
Yeah, I'll add a comment